--- kto1122-padavan-ng-org/trunk/user/www/n56u_ribbon_fixed/Advanced_Services_Content.asp	2023-10-09 14:55:04.569741112 +0200
+++ kto1122-padavan-ng/trunk/user/www/n56u_ribbon_fixed/Advanced_Services_Content.asp	2023-10-09 02:32:44.540541357 +0200
@@ -34,6 +34,7 @@ $j(document).ready(function() {
 	init_itoggle('tor_enable', change_tor_enabled);
 	init_itoggle('privoxy_enable', change_privoxy_enabled);
 	init_itoggle('dnscrypt_enable', change_dnscrypt_enabled);
+	init_itoggle('ttyd_enable', change_ttyd_enabled);
 });
 
 </script>
@@ -41,6 +42,10 @@ $j(document).ready(function() {
 
 <% login_state_hook(); %>
 <% openssl_util_hook(); %>
+var lan_ipaddr = '<% nvram_get_x("", "lan_ipaddr_t"); %>';
+var http_proto = '<% nvram_get_x("", "http_proto"); %>';
+var http_port = '<% nvram_get_x("", "http_lanport"); %>';
+var https_port = '<% nvram_get_x("", "https_lport"); %>';
 
 function initial(){
 	show_banner(1);
@@ -107,6 +112,12 @@ function initial(){
 		showhide_div('row_dnscrypt_options', 0);
 	}else
 		change_dnscrypt_enabled();
+
+	if(found_app_ttyd()){	
+		$("tbl_ttyd").style.display = "";
+	} else {
+		change_ttyd_enabled();
+	}
 }
 
 function applyRule(){
@@ -326,6 +337,19 @@ function change_crond_enabled(){
 		v = 0;
 	textarea_crond_enabled(v);
 }
+
+function change_ttyd_enabled(){
+	var v = document.form.ttyd_enable[0].checked;
+	showhide_div('ttyd_webui', v);
+	showhide_div('ttyd_port', v);
+}
+function on_ttyd_link(){
+	var ttyd_url="http";
+	var http_url=lan_ipaddr;
+	ttyd_url+="://"+http_url+":"+"<% nvram_get_x("","ttyd_port"); %>";
+	window_ttyd = window.open(ttyd_url, "ttyd");
+	window_ttyd.focus();
+}
 </script>
 <style>
     .caption-bold {
@@ -717,6 +741,36 @@ function change_crond_enabled(){
 					</tr>
                                     </table>
 
+                                    <table width="100%" id="tbl_ttyd" cellpadding="4" cellspacing="0" class="table" style="display:none;">
+                                        <tr>
+                                            <th colspan="2" style="background-color: #E3E3E3;"><#Adm_Svc_ttyd_setup#></th>
+                                        </tr>
+                                        <tr id="div_ttyd">
+                                            <th width="50%"><#Adm_Svc_ttyd_enable#></th>
+                                            <td colspan="2">
+                                                <div class="main_itoggle">
+                                                    <div id="ttyd_enable_on_of">
+                                                        <input type="checkbox" id="ttyd_enable_fake" <% nvram_match_x("", "ttyd_enable", "1", "value=1 checked"); %><% nvram_match_x("", "ttyd_enable", "0", "value=0"); %>>
+                                                    </div>
+                                                </div>
+                                                <div style="position: absolute; margin-left: -10000px;">
+                                                    <input type="radio" name="ttyd_enable" id="ttyd_enable_1" class="input" value="1" onclick="change_ttyd_enabled();" <% nvram_match_x("", "ttyd_enable", "1", "checked"); %>/><#checkbox_Yes#>
+                                                    <input type="radio" name="ttyd_enable" id="ttyd_enable_0" class="input" value="0" onclick="change_ttyd_enabled();" <% nvram_match_x("", "ttyd_enable", "0", "checked"); %>/><#checkbox_No#>
+                                                </div>
+                                            </td>
+                                        </tr>
+                                        <tr id="ttyd_port"> <th width="50%"><#Adm_Svc_ttyd_port#></th>
+                                            <td>
+                                                <input type="text" maxlength="6" class="input" size="15" name="ttyd_port" style="width: 145px" value="<% nvram_get_x("","ttyd_port"); %>" />
+                                            </td>
+                                        </tr>
+                                        <tr id="ttyd_webui">
+                                            <td>
+                                                <a href="javascript:on_ttyd_link();" id="web_ttyd_link">ttyd Web Shell</a>
+                                            </td>
+                                        </tr>
+                                    </table>
+
                                     <table width="100%" cellpadding="4" cellspacing="0" class="table">
                                         <tr>
                                             <th colspan="2" style="background-color: #E3E3E3;"><#Adm_System_misc#></th>
